### DISCUSSION TITLE
CyberPower UPS status display not correct

### DISCUSSION THREAD

**Discussion by Grumpy-Old-Engineer** [nutdotnet/WinNUT-Client#189] [Q&A] [answered]
WinNUT Client isn't showing the correct Output Voltage for my CyberPower UPS -shows 137 when it should be about 120.  Initially I thought I had a Battery Voltage issue also, but that turned out to be due to a bad Battery Voltage range on the Calibration tab.

![WinNUT_Display_r1](https://github.com/user-attachments/assets/90b4937f-1394-410f-b416-9996d1b6627f)

UPS (CPS/CP1500PFCLCDa/Unknown)
battery.charge (Description unavailable) : 100
battery.charge.low (Description unavailable) : 10
battery.charge.warning (Description unavailable) : 20
battery.mfr.date (Description unavailable) : CPS
battery.runtime (Description unavailable) : 3525
battery.runtime.low (Description unavailable) : 300
battery.type (Description unavailable) : PbAcid
battery.voltage (Description unavailable) : 27.3
battery.voltage.nominal (Description unavailable) : 24
device.mfr (Description unavailable) : CPS
device.model (Description unavailable) : CP1500PFCLCDa
device.serial (Description unavailable) : CXXKS2001121
device.type (Description unavailable) : ups
driver.name (Description unavailable) : usbhid-ups
driver.parameter.pollfreq (Description unavailable) : 30
driver.parameter.pollinterval (Description unavailable) : 5
driver.parameter.port (Description unavailable) : auto
driver.version (Description unavailable) : DSM6-2-25510-201118
driver.version.data (Description unavailable) : CyberPower HID 0.3
driver.version.internal (Description unavailable) : 0.38
input.voltage (Description unavailable) : 121.0
input.voltage.nominal (Description unavailable) : 120
output.voltage (Description unavailable) : 137.0
ups.beeper.status (Description unavailable) : disabled
ups.delay.shutdown (Description unavailable) : 20
ups.delay.start (Description unavailable) : 30
ups.load (Description unavailable) : 13
ups.mfr (Description unavailable) : CPS
ups.model (Description unavailable) : CP1500PFCLCDa
ups.productid (Description unavailable) : 0601
ups.realpower.nominal (Description unavailable) : 1000
ups.serial (Description unavailable) : CXXKS2001121
ups.status (Description unavailable) : OL
ups.test.result (Description unavailable) : No test initiated
ups.timer.shutdown (Description unavailable) : -60
ups.timer.start (Description unavailable) : -60
ups.vendorid (Description unavailable) : 0764

[WinNUT-Client-2024-09-21.log](https://github.com/user-attachments/files/17082955/WinNUT-Client-2024-09-21.log)

**Comment by gbakeman** [comment]
Hi @Grumpy-Old-Engineer ,

Thanks for posting a new discussion. I was racking my brain last time trying to remember where I had seen this issue before... And it turns out it's exactly like #101. As the other user noted, this is a well-documented bug amongst the Network UPS Tools community, noting that it was fixed as of UPSd 2.8.0. It does appear that you're using Synology's software, and an old version at that. I'd suggest updating your DiskStation Management software if at all possible, and if not, you'll want to find another way of installing the first-party build of UPSd so that you can monitor your UPS with updated software.

I'm sorry I don't have a better answer for you, unfortunately WinNUT does depend a lot on the server that's presenting data to it.

**Comment by Grumpy-Old-Engineer** [comment]
Thanks for the help.  I've been running the latest in Synology's DSM 6.2.2.xxxxx path.  I tried to run a UPSd Docker image - it seems to start but it just won't stay up.  I have avoided DSM 7 because I heard too many "not good" things when it was released.  Based on your comment I did a little research and decided to risk the update.  I am now running the latest, DSM 7.2.2.72806, and the problem didn't go away.
![WinNUT_Display_r2](https://github.com/user-attachments/assets/58a93c6a-474d-425d-86f4-fac7eb684478)
[WinNUT_MIB_data_r2.txt](https://github.com/user-attachments/files/17088538/WinNUT_MIB_data_r2.txt)
[WinNUT-Client-2024-09-22.log](https://github.com/user-attachments/files/17088539/WinNUT-Client-2024-09-22.log)

**Reply by gbakeman** [reply]
Thanks for attempting the update, and I'm sorry it didn't fix the issue for you. I'm trying to see where Synology mentions what upstream version of NUT/upsd they're using, although I'm struggling to find it so I think we can only assume that they're still using an older version here. Can you troubleshoot the docker container further to at least get to the point where you can confirm that it presents the data correctly?

**Comment by Grumpy-Old-Engineer** [comment]
The Docker attempt was with DSM 6.2.2.  DSM 7 doesn't support Docker, so I tried something called Container Manager, but that gives me the same error that I got with DSM 6.2.2 and Docker.  I'm not a programmer, so competent debugging is a bit outside my skill set.  The attached are the log records from the last 2 times I tried to start the container - it always ends with the error about "chgrp : /dev/bus/usb: No such file or directory".  Since I can see /dev/bus/usb on the NAS I'm guessing that it's something in the container itself.
[upshift-nut-upsd1_log.txt](https://github.com/user-attachments/files/17091225/upshift-nut-upsd1_log.txt)

**Reply by gbakeman** [reply]
I'm sorry to say that this is starting to reach my limits of knowledge as well as I don't have much experience either with containers in Synology or running upsd in a container. Is there any chance you have another machine to test that can run upsd in a more traditional physical OS or even a VM, just to confirm that the issue would be fixed?

**Reply by Grumpy-Old-Engineer** [reply]
No.... Synology's DSM is almost, but not quite Linux, so I used to have a very old X58 PC with a 1st gen Core i7 where I ran Ubuntu for playing with things like this.  Something died and even the cheap engineer in me said it wasn't worth the time to try to fix.  This UPS issue is the first thing I've run up against since I scraped that old PC where I want a Linux box to play on.

So, my options seem to be:
1. Poke at Synology  (just submitted a service request)
2. Try to ask the container author for insight as to why it won't run  
3. Build a Linux sandbox machine and try to build my own container  (not just for this.....)
4. Deal with a funky "Output Voltage" display  (most likely)

**Reply by gbakeman** [reply, marked as answer]
I'm afraid that's what I'm seeing. I know you're not the only one with this problem, and I can't rule out an edge-case fix for this in WinNUT specifically - but I think the options you listed are the best ones at the moment.

**Reply by Grumpy-Old-Engineer** [reply]
If anyone is interested, I just heard back from Synology.  
    "I have received an update from our developers, and the NAS is using Network UPS Tools (NUT) with Driver 2.6."
     https://archive.synology.com/download/ToolChain/Synology%20NAS%20GPL%20Source/7.2-64570
I downloaded the file for my DS416Play (INTEL Celeron N3060 - Braswell) and poked around - the last entry in the change log is dated 5/31/2012.  My response back to Synology equates to "please update with the latest version".

Can we all say OMG!!!

**Reply by gbakeman** [reply]
Thank you for looking into this and posting back with results! I wonder what the chances are of Synology coming through with an update. Do you think you'll look further into the container idea?

**Reply by Grumpy-Old-Engineer** [reply]
1) I have no idea how responsive Synology will be.  In my opinion it's not a good business practice for them to ignore something this old if they don't want users bashing DSM now that the details of this extremely obsolete driver have been discovered.  Makes me wonder how many other drivers are out of date and being ignored by Synology.
2) I sent an email to an address I found buried the container's image history.  I'm waiting to see if anyone responds, but at least I haven't gotten a "no such address" bounce back.
